### PR TITLE
feat: add feature flag parameter to getAuthUrl

### PR DESCRIPTION
**Pull request by allisonc07**
Enable getAuthUrl to accept a `flags` parameter for enabling feature flags

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -78,6 +79,7 @@ response_type=code
 &state=0facda3319
 &make=TESLA
 &single_select=true
+&flags=country:DE
```
Comment (nkamidi) [inline doc/README.md:82]: Can you a couple more feature flag examples to "flags" parameter. Something like '&flags=country:DE%20test1:value1%20test2:value2'. Developers could understand that '%20" is html encoded value for space but we could put a note saying so. What are your thoughts about this?

**Review by nkamidi — APPROVED** [review]
Looks good to me. I will approve it but will follow up offline on my question.

File: src/sdk.js [hunk 2]
```diff
@@ -248,6 +248,9 @@ class Smartcar {
    * the vehicle with the specified VIN. See the
    * [API reference](https://smartcar.com/docs/api/#connect-match)
    * for more information.
+   * @param {String[]} [options.flags] - An optional space-separated list of feature
+   * flags that your application has early access to. Flags are colon separated for
+   * any suboptions they may have
```
Comment (sankethkatta) [inline src/sdk.js:253]:
```suggestion
   * @param {String[]} [options.flags] - An optional space-separated list of feature
   * flags that your application has early access to. 
```

**Review by sankethkatta — CHANGES_REQUESTED** [review]

File: doc/README.md (hunk 1 above)
Comment (allisonc07) [reply doc/README.md:82]: I will add another example! I will keep it un-encoded though like the example for the `scope` parameter a couple of lines above it.

**Review by sankethkatta — APPROVED** [review]

**Pull request merged**